src/yb/yql/cql/cqlserver/cql_processor.cc

```diff
@@ -505,7 +503,7 @@ CQLResponse* CQLProcessor::ProcessResult(const ExecutedResult::SharedPtr& result
                   "Provided username " + params.username + " and/or password are incorrect");
             }
             const auto& saved_hash = salted_hash_value.string_value();
-            if (bcrypt_checkpw(params.password.c_str(), saved_hash.c_str())) {
+            if (!service_impl_->CheckPassword(params.password, saved_hash)) {
```

Comment:
We need a flag check here and leave the original code as the default value:

```
bool valid_password = false;
if (FLAGS_use_password_hash_cache) {
  valid_password = service_impl_->CheckPassword(params.password, saved_hash)
} else {
  valid_password = bcrypt_checkpw(params.password.c_str(), saved_hash.c_str());
}
if (valid_password) {
```

Reply:
Where should we define these flags?

Reply:
Additionally, which binaries need to declare these flags?